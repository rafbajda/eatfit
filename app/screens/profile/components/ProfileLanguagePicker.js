import React from 'react';
import { deepBlue } from '../../../shared/constants/colors';
import { Picker, Icon, Grid, Col } from 'native-base';
import styles, {
    ProfileLanguageRow,
    ProfilePickerColumn
} from '../styles/profileLanguagePickerStyles';

const ProfileLanguagePicker = props => {
    const { languages, placeholderStyle, userLanguage, setUserLanguage } = {
        ...props,
        ...styles
    };
    console.log('props: ', props);
    const PickerItems = languages.map(language => (
        <Picker.Item
            label={language.label}
            value={language.id}
            key={language.id}
        />
    ));

    return (
        <Grid>
            <ProfileLanguageRow>
                <Col />
                <Col />
                <ProfilePickerColumn>
                    <Picker
                        mode="dropdown"
                        iosIcon={<Icon name="arrow-down" />}
                        placeholderStyle={placeholderStyle}
                        placeholderIconColor={deepBlue}
                        selectedValue={userLanguage}
                        onValueChange={language => setUserLanguage(language)}
                    >
                        {PickerItems}
                    </Picker>
                </ProfilePickerColumn>
            </ProfileLanguageRow>
        </Grid>
    );
};

export default ProfileLanguagePicker;
