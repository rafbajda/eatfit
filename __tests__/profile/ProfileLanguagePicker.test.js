import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import ProfileLanguagePicker from '../../app/screens/profile/components/ProfileLanguagePicker';

describe('ProfileLanguagePicker snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<ProfileLanguagePicker />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
