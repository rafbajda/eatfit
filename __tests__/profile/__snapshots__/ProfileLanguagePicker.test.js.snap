// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLanguagePicker snapshot renders component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    className="sc-bdVaJa bqydcF"
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    />
    <View
      className="sc-bwzfXH bSsNQu"
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            note={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#a7a7a7",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "lineHeight": null,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                undefined,
                Object {
                  "color": "#dfe3ee",
                },
              ]
            }
            uppercase={false}
          />
          <Text />
        </View>
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          transparent={false}
          visible={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
                "height": 1334,
              }
            }
          >
            <View
              onLayout={[Function]}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F8F8F8",
                    "borderBottomColor": "#a7a6ab",
                    "borderBottomWidth": 0.5,
                    "elevation": 3,
                    "flexDirection": "row",
                    "height": 64,
                    "justifyContent": "center",
                    "left": 0,
                    "paddingLeft": 6,
                    "paddingRight": 10,
                    "paddingTop": 18,
                    "right": 0,
                    "shadowColor": undefined,
                    "shadowOffset": undefined,
                    "shadowOpacity": undefined,
                    "shadowRadius": undefined,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": null,
                        "backgroundColor": "transparent",
                        "borderBottomWidth": null,
                        "borderColor": null,
                        "borderLeftWidth": null,
                        "borderRadius": 5,
                        "borderRightWidth": null,
                        "borderTopWidth": null,
                        "borderWidth": undefined,
                        "elevation": 0,
                        "flexDirection": "row",
                        "height": 45,
                        "justifyContent": "space-between",
                        "marginLeft": 3,
                        "opacity": 1,
                        "paddingBottom": 6,
                        "paddingLeft": 4,
                        "paddingRight": 6,
                        "paddingTop": 6,
                        "shadowColor": null,
                        "shadowOffset": null,
                        "shadowOpacity": null,
                        "shadowRadius": null,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#007aff",
                          "fontFamily": "System",
                          "fontSize": 17,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "paddingLeft": 2,
                          "paddingRight": undefined,
                          "top": 1,
                        }
                      }
                      uppercase={false}
                    >
                      Back
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 17,
                        "fontWeight": "700",
                        "marginLeft": undefined,
                        "paddingLeft": 4,
                        "paddingTop": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Select One
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "alignSelf": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "flex-end",
                    }
                  }
                />
              </View>
            </View>
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
        </Modal>
      </View>
    </View>
  </View>
</View>
`;
